### PR TITLE
Fix machine rain exploding in biomes that does not have visual rains

### PR DESCRIPTION
**Pull request by Glease**
Fix https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/10174

### CODE REVIEW

**Comment by Chaoschaot232** [comment]
Thanks for this fast fix @Glease 👍

File: src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java [hunk 1]
```diff
@@ -219,6 +220,11 @@ public void dischargeItem(ItemStack aStack) {
         increaseStoredEnergyUnits(GT_ModHandler.dischargeElectricItem(aStack, (int) Math.min(Integer.MAX_VALUE, getEUCapacity() - getStoredEU()), (int) Math.min(Integer.MAX_VALUE, mMetaTileEntity.getInputTier()), false, false, false), true);
     }
 
+    protected boolean isRainPossible() {
+        BiomeGenBase biome = getBiome();
+        return biome.rainfall > 0 && (!biome.getEnableSnow() && !biome.canSpawnLightningBolt());
```
Comment (GlodBlock) [inline src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java:225]: why snow and thunder don't count as rain weather?

Comment (Glease) [reply src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java:225]: It's missing a ! out of the parenthesis. Probably lost during refactors...

Comment (Glease) [reply src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java:225]: For why these two conditions are used, see net.minecraft.client.renderer.EntityRenderer#renderRainSnow.

**Review by GlodBlock — APPROVED** [review]

**Pull request merged**